### ISSUE TITLE
Mod crashes game upon start

### ISSUE THREAD

**Issue by Alatarius** [DustyBagel/hopper_compat#3]
As soon as I tried to start minetest_game it crashed with the following error. Unable to start game. [edit] This is after the new update.

ModError: Failed to load and run script from D:\Games\Minetest\bin\..\mods\hopper_compat\init.lua:
D:\Games\Minetest\bin\..\mods\hopper_compat\init.lua:50: bad argument #1 to 'pairs' (table expected, got nil)
stack traceback:
	[C]: in function 'pairs'
	D:\Games\Minetest\bin\..\mods\hopper_compat\init.lua:50: in function 'remove_hopper_support'
	D:\Games\Minetest\bin\..\mods\hopper_compat\init.lua:65: in main chunk
Check debug.txt for details.

**Comment by DustyBagel** [comment]
Which release are you using? I have encountered issues with v2.0.0 and v2.0.1 but I never have had it crash. I am currently fixing the issues with v2.0.0 and v2.0.1.  Try release v2.0.2 and see if you are still encountering this issue.

I can't replicate the crash. Do you have any other mods installed other than the default minetest_game mods, hopper, hopper_compat and connected_chests? What operating system are you using? What minetest version are you running on?

As a temporary fix you could enable the protected chest support setting. When this is enabled, the hoppers can interact with locked nodes and the remove_hopper_support function is not used.

**Comment by DustyBagel** [comment]
I can't replicate the crash. Do you have any other mods installed other than the default minetest_game mods, hopper, hopper_compat and connected_chests? What operating system are you using? What minetest version are you running on?

I ran hopper_compat version 2.0.2 in a new world on minetest_game with only hoppers, hopper_compat and connected_chests enabled on the stable release minetest version 5.8.0, and I didn't get the crash message.

As a temporary fix you could enable the protected chest support setting. When this is enabled, the hoppers can interact with locked nodes and the remove_hopper_support function is not used.

**Comment by Alatarius** [comment]
I'm using whatever version is on ContentDB. As for minetest version I'm using the newest release of 5.8.0. I will try the newest update that's available in minetest as I see it has updated.

[edit] I am running on Windows 11 and I am playing singleplayer with no locked chests. 

As for mods, here is my list:

3d_apple, 3d_armor, ambience, amethyst_new, animalworld, anvil, area_containers, artdeco, auroras, awards, bakedclay, banner, basic_materials, beds, biofuel, biomeinfo, biome_lib, bonemeal, bows, bricks_plus, bucket, bucket_wooden, builtin_item, carpets, castle_farming, castle_gates, castle_lighting, castle_masonry, castle_shields, castle_storage, castle_tapestries, castle_weapons, connected_chests, controls, cottages, cracked_castle, craftguide, darkage, decorations_sea, deepslate, diggicontrol, digilines, digistuff, digtron, display_modpack, doc, doc_basics, doc_encyclopedia, doc_identifier, doc_items, doors, drinks, ebiomes, ethereal, etherium_stuff, everness, factory_bridges, farming, flowerpot, geodes_lib, glass_stained, handle_schematics, hangglider, hopper, hopper_compat, illumination, itemframes, item_drop, leads, leather_armor, livingcaves, livingcavesmobs, livingdesert, livingfloatlands, livingjungle, livingnether, loot, lwcomponents, lwscratch, marinara, marinaramobs, mesecons, mesecons_carts, mesecons_x, micupack, minecart, mobkit, mobs, mobs_monster, mobs_skeletons, mobs_sky, mobs_water, mob_horse, modlib, modname_tooltip, modutil, moreblocks, moredecorations, morelights, moreores, moretorches, moretrees, music_modpack, nativevillages, naturalbiomes, nether, new_campfire, obsidianstuff, paintings_gallery, paintings_lib, pipeworks, pipunhole, plaster, powerbanks, redef, regrowing_fruits, rnd, ropes, scaffolding, servercleaner, shallow_ores, signs_lib, simple_furniture, skinsdb, slats, stainedglass, stripped_tree, sugar, swamp, technic_plus, technic_recipes, techpack, tga_encoder, torch_bomb, tpad, ts_furniture, tt, tubelib2, unifieddyes, unified_inventory, variety, visible_sneak, void_chest, we_undo, wilhelminessounds, wine, woodcutting, xdecor, xmaps, x_enchanting, x_farming, factory

**Comment by DustyBagel** [comment]
I believe v2.0.3 should be up now.

**Comment by Cebobilff** [comment]
Make on that is an unblocked website and post it in your notes

**Comment by Alatarius** [comment]
I downloaded the mod from here on github and I still get the same error

**Comment by DustyBagel** [comment]
I downloaded a new install of minetest, minetest_game hopper, hopper_compat and connected_chests and I was able to recreate the crash. I will investigate this and let you know when I find a fix.

[Edit] I found the problem. You have the version of the hopper mod by tenplus1. hopper_compat only works with facedeer's version of the hopper mod. I didn't know they were different. I will have to tell users about that. Try installing that version of the mod from github or from the content database but should you download it from there if release v2.0.4 is on the content database and see if it works.
